woocommerce/woocommerce
# Block Checkout: Fix default shipping selection on toggle pickup

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Fixes a regression from https://github.com/woocommerce/woocommerce/pull/48370 noticed by @nielslange whereby toggling between shipping and pickup would not reselect a shipping method automatically. 

The reason for this is unlike with pickup locations, the component forces the default state to be the first shipping rate in the array. Because of this, the event is not pushed up to the data store.

By defaulting to a blank default shipping rate, the mounting code can select the shipping rate instead and push the change upward. This matches up pickup locations do it.

Additionally, there was an issue where this notice would flash up when changing between shipping and pickup:

![Screenshot 2024-07-19 at 14 00 27](https://github.com/user-attachments/assets/e7ad1901-012e-49dd-9d23-73e77b0a06b4)

This can be fixed by removing the notice during calculations.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Setup some shipping methods and some local pickup locations under WooCommerce > Settings > Shipping. Additionally, create a zone for the US with no shipping methods. We'll use this later.
2. Go to checkout.
3. Using a non-US address,Toggle between local pickup and shipping. Ensure when you toggle, the first shipping rate or pickup location is automatically selected and totals are updated as expected. Also ensure that the notice "Totals will be recalculated when a valid shipping method is selected." does not become visible.
4. Switch to shipping and enter a US address (ensure you select a state also in the form as there is an unrelated issue with the select boxes preventing default state from being selected).
5. Check that the "Totals will be recalculated when a valid shipping method is selected." method appears and there are no shipping methods available.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
